lib/tiddle/model_name.rb

```diff
@@ -1,11 +1,17 @@
 module Tiddle
   class ModelName
     def with_underscores(model)
-      model.model_name.to_s.underscore.upcase
+      colon_to_underscore(model).underscore.upcase
     end
 
     def with_dashes(model)
       with_underscores(model).dasherize
     end
+
+    private
+
+    def colon_to_underscore(model)
+      model.model_name.to_s.gsub(':', '_')
```

Comment:
**nitpick**: Doesn't `model_name` already return a string?

**nitpick**: You can use `tr` instead of `gsub`. Could be a few nanoseconds faster.

Reply:
**issue**: The spec is testing with `X-NAMESPACE--NAMESPACED-USER-EMAIL` (dashes) but this one uses `_` (underscores). I guess something is wrong?

Reply:
`nitpick: Doesn't model_name already return a string?`
* I can test this

`nitpick: You can use tr instead of gsub. Could be a few nanoseconds faster.`
* I read about it and it makes sense.

`The spec is testing with X-NAMESPACE--NAMESPACED-USER-EMAIL (dashes) but this one uses _ (underscores). I guess something is wrong?`
* That's correct, the HTTP header uses `-` and not `_`, ActiveRecord replaces both.